cpp/src/arrow/util/future.h

```diff
@@ -573,6 +573,15 @@ class ARROW_MUST_USE_TYPE Future {
   FRIEND_TEST(FutureRefTest, HeadRemoved);
 };
 
+namespace internal {
```

Comment:
I don't see this used.  What is the intent?  There is already `Future<T>::status`

Reply:
Probably for use within `ASSERT_OK`, but we already have specific macros for futures.

Reply:
I'll revert this, then